shell/platform/linux/fl_binary_messenger.cc

```diff
@@ -148,16 +179,15 @@ G_MODULE_EXPORT gboolean fl_binary_messenger_send_response(
     GError** error) {
   g_return_val_if_fail(FL_IS_BINARY_MESSENGER(self), FALSE);
   g_return_val_if_fail(response_handle != nullptr, FALSE);
-
-  // Take reference to ensure it is freed
-  g_autoptr(FlBinaryMessengerResponseHandle) owned_response_handle =
-      response_handle;
+  g_return_val_if_fail(response_handle->messenger == self, FALSE);
+  g_return_val_if_fail(response_handle->response_handle != nullptr, FALSE);
```

Comment:
Will this log by default? I'd like to always log an error message if someone tries to double-respond, to increase the chances of it being noticed.

Reply:
Yes, it will.